src/plugins/data/common/data_views/data_views/data_view.ts

```diff
@@ -0,0 +1,393 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import _, { each, reject } from 'lodash';
+import { FieldFormatsContentType, SavedObjectsClientCommon } from '../..';
+import { DuplicateField } from '../../../../opensearch_dashboards_utils/common';
+
+import { SerializedFieldFormat } from '../../../../expressions/common';
+import {
+  FieldFormatNotFoundError,
+  IDataViewFieldType,
+  IDataView,
+  OPENSEARCH_FIELD_TYPES,
+  OSD_FIELD_TYPES,
+} from '../..';
+import { FieldFormat, FieldFormatsStartCommon } from '../../field_formats';
+import { IDataViewFieldList, DataViewField, dataViewFieldList as fieldList } from '../fields';
+import {
+  DataViewFieldMap,
+  DataViewSpec,
+  DataViewSavedObjectReference,
+  DataViewSourceFilter,
+  DataViewTypeMeta,
+} from '../types';
+import { flattenHitWrapper } from './flatten_hit';
+import { formatHitProvider } from './format_hit';
+
+interface DataViewDeps {
+  spec?: DataViewSpec;
+  savedObjectsClient: SavedObjectsClientCommon;
+  fieldFormats: FieldFormatsStartCommon;
+  shortDotsEnable: boolean;
+  metaFields: string[];
+}
+
+interface SavedObjectBody {
+  title?: string;
+  displayName?: string;
+  description?: string;
+  timeFieldName?: string;
+  intervalName?: string;
+  fields?: string;
+  sourceFilters?: string;
+  fieldFormatMap?: string;
+  typeMeta?: string;
+  type?: string;
+}
+
+type FormatFieldFn = (
+  hit: Record<string, any>,
+  fieldName: string,
+  type?: FieldFormatsContentType
+) => any;
+
+const DATA_SOURCE_REFERNECE_NAME = 'dataSource';
+export class DataView implements IDataView {
```

Comment:
Can we have a comment on the intended purpose of this and communicate what this will be called in the UI to help avoid confusion in the future?